flaky test: TestUserNoEffectiveCapabilitiesNetBindService (open /proc/sys/net/ipv4/ip_unprivileged_port_start: no such file or directory)

```
[2020-06-21T02:15:25.118Z] === Failed

[2020-06-21T02:15:25.118Z] === FAIL: ppc64le.integration-cli TestDockerSuite/TestUserNoEffectiveCapabilitiesNetBindService (1.04s)

[2020-06-21T02:15:25.118Z]     --- FAIL: TestDockerSuite/TestUserNoEffectiveCapabilitiesNetBindService (1.04s)

[2020-06-21T02:15:25.118Z]         docker_cli_run_unix_test.go:1256: assertion failed: 

[2020-06-21T02:15:25.118Z]             Command:  /usr/local/cli/docker run --user 1000:1000 --sysctl net.ipv4.ip_unprivileged_port_start=1024 syscall-test socket-test

[2020-06-21T02:15:25.118Z]             ExitCode: 127

[2020-06-21T02:15:25.118Z]             Error:    exit status 127

[2020-06-21T02:15:25.118Z]             Stdout:   

[2020-06-21T02:15:25.118Z]             Stderr:   /usr/local/cli/docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"write sysctl key net.ipv4.ip_unprivileged_port_start: open /proc/sys/net/ipv4/ip_unprivileged_port_start: no such file or directory\"": unknown.

[2020-06-21T02:15:25.118Z]             time="2020-06-21T01:23:23Z" level=error msg="error waiting for container: context canceled" 

[2020-06-21T02:15:25.118Z]             

[2020-06-21T02:15:25.118Z]             

[2020-06-21T02:15:25.118Z]             Failures:

[2020-06-21T02:15:25.118Z]             ExitCode was 127 expected 1

[2020-06-21T02:15:25.118Z]             Expected stderr to contain "Permission denied"
```

https://ci-next.docker.com/public/blue/organizations/jenkins/moby/detail/master/541/pipeline

--- a/integration-cli/docker_cli_run_unix_test.go
+++ b/integration-cli/docker_cli_run_unix_test.go
@@ -1245,6 +1245,14 @@ func (s *DockerSuite) TestUserNoEffectiveCapabilitiesSetgid(c *testing.T) {
 
 // TODO CAP_SETPCAP
 
+// sysctlExists checks if a sysctl exists; runc will error if we add any that do not actually
+// exist, so do not add the default ones if running on an old kernel.
+func sysctlExists(s string) bool {
+	f := filepath.Join("/proc", "sys", strings.Replace(s, ".", "/", -1))
+	_, err := os.Stat(f)
+	return err == nil
+}
+
 func (s *DockerSuite) TestUserNoEffectiveCapabilitiesNetBindService(c *testing.T) {
 	testRequires(c, DaemonIsLinux, testEnv.IsLocalDaemon)
 	ensureSyscallTest(c)
@@ -1253,12 +1261,22 @@ func (s *DockerSuite) TestUserNoEffectiveCapabilitiesNetBindService(c *testing.T
 	dockerCmd(c, "run", "syscall-test", "socket-test")
 	// test that non root user does not have default capability CAP_NET_BIND_SERVICE
 	// as we allow this via sysctl, also tweak the sysctl back to default
-	icmd.RunCommand(dockerBinary, "run", "--user", "1000:1000", "--sysctl", "net.ipv4.ip_unprivileged_port_start=1024", "syscall-test", "socket-test").Assert(c, icmd.Expected{
+	args := []string{"run", "--user", "1000:1000"}
+	if sysctlExists("net.ipv4.ip_unprivileged_port_start") {
+		args = append(args, "--sysctl", "net.ipv4.ip_unprivileged_port_start=1024")
+	}
+	args = append(args, "syscall-test", "socket-test")
+	icmd.RunCommand(dockerBinary, args...).Assert(c, icmd.Expected{
 		ExitCode: 1,
 		Err:      "Permission denied",
 	})
 	// test that root user can drop default capability CAP_NET_BIND_SERVICE
-	icmd.RunCommand(dockerBinary, "run", "--cap-drop", "net_bind_service", "--sysctl", "net.ipv4.ip_unprivileged_port_start=1024", "syscall-test", "socket-test").Assert(c, icmd.Expected{
+	args = []string{"run", "--cap-drop", "net_bind_service"}
+	if sysctlExists("net.ipv4.ip_unprivileged_port_start") {
+		args = append(args, "--sysctl", "net.ipv4.ip_unprivileged_port_start=1024")
+	}
+	args = append(args, "syscall-test", "socket-test")
+	icmd.RunCommand(dockerBinary, args...).Assert(c, icmd.Expected{
 		ExitCode: 1,
 		Err:      "Permission denied",
 	})

